src/Stack/Build.hs

```diff
@@ -128,6 +132,65 @@ allLocal =
     Map.elems .
     planTasks
 
+-- | See https://github.com/commercialhaskell/stack/issues/1198.
+warnIfExecutablesWithSameNameCouldBeOverwritten
+    :: MonadLogger m => [LocalPackage] -> Plan -> m ()
+warnIfExecutablesWithSameNameCouldBeOverwritten locals plan = do
+    let localDuplicateExes = Map.filter (\ps -> length ps > 1) (localExes locals)
```

Comment:
Actually, nevermind on the `exesToBuild` / `plannedExes` stuff, maybe those should just be deleted.  See my comment [here](https://github.com/commercialhaskell/stack/pull/1373#issuecomment-158235789)